### DISCUSSION TITLE
Open single toggle instead of multiple

### DISCUSSION THREAD

**Discussion by asmsaiff** [alpinejs/alpine#3388] [1. Help]
Hi there,

I've a navigation in my web page where each menu item will trigger a box on click. Look here - https://prnt.sc/aCWZG2Uc-ITE

How can I trigger only one element at a time? I mean when I'll click on the ```Product Family``` then other two will be collapsed. Now It's triggering all of them at the same time.

Thanks.

**Comment by ekwoka** [comment]
you would need to show code.

But you can have each have their own state, and use a setter to do it.

**Reply by asmsaiff** [reply]
Hi, @ekwoka  Here is my code for the toggle -

```html
<!-- First Toggle -->
<div class="relative" x-data="{ open: false }">
    <button type="button" x-on:click="open = ! open">
        <h3>Product Family</h3>
        <p class="text-sm text-cyan-500">lorem ipsum dolor</p>
    </button>
    <ul class="w-48 absolute bottom-14 p-2 text-sm flex gap-4 flex-col bg-gray-900" x-show.important="open">
        <li>
            <a href="">Product Family 1</a>
        </li>
        <li>
            <a href="">Product Family 2</a>
        </li>
        <li>
            <a href="">Product Family 3</a>
        </li>
    </ul>
</div>

<!-- Second Toggle -->
<div class="relative" x-data="{ open: false }">
    <button type="button" x-on:click="open = ! open">
        <h3>Volume</h3>
        <p class="text-sm text-cyan-500">lorem ipsum dolor</p>
    </button>
    <ul class="absolute bottom-14 p-2 text-sm w-48 flex gap-4 flex-col bg-gray-900" x-show.important="open">
        <li>
            <a href="">Volume 1</a>
        </li>
        <li>
            <a href="">Volume 2</a>
        </li>
        <li>
            <a href="">Volume 3</a>
        </li>
    </ul>
</div>
```

**Reply by ekwoka** [reply]
And this opens all of them when you click on one of them?

Or you mean it allows all of them to be open at once?

**Reply by asmsaiff** [reply]
It allows all of them to be open at once. I want it will open only one at a time.

**Reply by ekwoka** [reply]
Then i'd suggest refactoring to

```html
<div x-data="{ tabs: [] }"> 
  <div x-data="{ get isOpen() { return tabs[0] }, set isOpen(val) { tabs = tabs.map((_, i) => i === 0 ? val : false) } }">
     <button @click="isOpen = !isOpen">
     <div x-show="isOpen">
  </div>
  ....repeat
</div>
```

Ideally in your templating you're looping over the menu so you can use that to inject the indexes, or extract the repetiving inner component to a data constructor you can repeat.

**Reply by asmsaiff** [reply]
Thanks for sharing your idea. I'm trying with this.
